### PR TITLE
Feature: overview page showing all decisions that have a regulation in them

### PR DESCRIPTION
**Pull request by usrtim**
https://user-images.githubusercontent.com/115715476/235473217-bcdeea97-0d45-4bf0-887d-93edc8ca19c3.mp4

### Relevant Jira tickets

https://binnenland.atlassian.net/browse/GN-4168

### dependencies

needs https://github.com/lblod/app-gn-publicatie/pull/22 on the backend

### setup

needs a working GN + publication stack sync. Below is the relevant parts of my setup.
NOTE: I don't run the dev file for GN.

my override for publication backend:

<details>
<summary>publication override</summary>

```yml
version: "3.4"

services:
  identifier:
    restart: "no"
    networks:
      default:
        aliases:
          - publication_identifier

  published-resource-consumer:
    restart: "no"
    environment:
      NODE_ENV: development
      SERVICE_NAME: 'published-resource-consumer'
      SYNC_BASE_URL: 'http://gn-identifier'
      INGEST_INTERVAL: 5000
    networks:
      - default
      - shared-gn-pub

networks:
  shared-gn-pub:
    name: shared-gn-pub
```

</details>

<details>
<summary>GN override</summary>

```yaml
version: "3.4"
x-logging: &default-logging
  driver: "json-file"
  options:
    max-size: "10m"
    max-file: "3"
services:
  identifier:
    ports:
      - "4302:80"
    environment:
      SESSION_COOKIE_SECURE: "false"
    networks:
      default:
      shared-gn-pub:
        aliases:
          - gn-identifier
    restart: no
networks:
  shared-gn-pub:
    name: shared-gn-pub

```
</details>

### how to test
- publish a decision which includes a regulatory statement (called "uittreksel" or "extract") from GN
- in a few seconds, that decision should be visible in both the "zittingen" section, and in the "reglementen" section

### challenges

unsure about rdfa correctness

### CODE REVIEW

**Comment by usrtim** [comment]
> Just seeing the video you can see that there are some problems with the styling of the button when the title and description are missing

@lagartoverde Yes that's a quick fix(I have fx and will push soon). Can you please look into the functionality and let me know If I'm missing something

File: app/controllers/bestuurseenheid/reglementen/index.js
Comment (abeforgit) [inline app/controllers/bestuurseenheid/reglementen/index.js:19]: don't overwrite the model property on a controller...
https://guides.emberjs.com/release/routing/query-params/ please read this again

File: app/controllers/bestuurseenheid/reglementen/index.js
Comment (abeforgit) [inline app/controllers/bestuurseenheid/reglementen/index.js:20]: why don't you do this in the model hook? https://guides.emberjs.com/release/routing/query-params/

File: app/controllers/bestuurseenheid/reglementen/index.js
Comment (abeforgit) [inline app/controllers/bestuurseenheid/reglementen/index.js:14]: the model property on a controller is automatically set, and is always tracked. It gets the resolved return value of the model hook in the route class

**Review by abeforgit — CHANGES_REQUESTED** [review]

File: app/controllers/bestuurseenheid/reglementen/index.js
Comment (usrtim) [reply app/controllers/bestuurseenheid/reglementen/index.js:19]: I got this practice from here: https://github.com/lblod/frontend-gelinkt-notuleren-publicatie/blob/master/app/controllers/bestuurseenheid/index.js#L115 . I just wanted to stay consistent with other controllers. I will have a look now how we can do it better

File: app/controllers/bestuurseenheid/reglementen/index.js
Comment (usrtim) [reply app/controllers/bestuurseenheid/reglementen/index.js:14]: the same also here, I tried to stay consistent with other controllers: https://github.com/lblod/frontend-gelinkt-notuleren-publicatie/blob/master/app/controllers/bestuurseenheid/index.js#L71 . I will have a look now

File: app/controllers/bestuurseenheid/reglementen/index.js
Comment (usrtim) [reply app/controllers/bestuurseenheid/reglementen/index.js:20]: The same also here: https://github.com/lblod/frontend-gelinkt-notuleren-publicatie/blob/master/app/controllers/bestuurseenheid/index.js#L106. I will take a look now into this

File: app/controllers/bestuurseenheid/zitting/uittreksels/index.js
Comment (lagartoverde) [inline app/controllers/bestuurseenheid/zitting/uittreksels/index.js:11]: The whole point of using this was to not use the .firstObject because it's deprecated, use [0] instead :).
Also try to avoid using the .get('besluiten') as it's also deprecated, you might need include them in the store query of the page if you are getting an error that behandelingVanAgendapunt is a proxy

**Review by lagartoverde — CHANGES_REQUESTED** [review]
Also you still have 2 footers on the reglementen page

File: app/components/uittreksel-overview-item.hbs
Comment (lagartoverde) [inline app/components/uittreksel-overview-item.hbs:1]: I would rename getBesluit to fetchBesluit now

**Review by lagartoverde — CHANGES_REQUESTED** [review]
You still have the duplicate footer problem.
Once that's solved we can merge this

**Comment by usrtim** [comment]
> You still have the duplicate footer problem. Once that's solved we can merge this

@lagartoverde please check it now

**Review by lagartoverde — APPROVED** [review]
Looks good to me :)

File: app/controllers/bestuurseenheid/reglementen/index.js
Comment (abeforgit) [inline app/controllers/bestuurseenheid/reglementen/index.js:7]: these are not meetings, why are they called that
and in any case, you always have access to the route model in the template with `@model`, so this getter is a bit redundant (I'm not against it though! It can be nice to give a more descriptive name to the model, but if you do so, it should at least be correct...)

File: app/routes/bestuurseenheid/reglementen/index.js
Comment (abeforgit) [inline app/routes/bestuurseenheid/reglementen/index.js:38]: query params are supposed to be defined on the controller:
https://guides.emberjs.com/v4.8.0/routing/query-params/

what's the reason for tacking them onto the ember data model?

File: app/routes/bestuurseenheid/reglementen/index.js
Comment (abeforgit) [inline app/routes/bestuurseenheid/reglementen/index.js:54]: why do this?
why not just return `model` directly from the modelhook, and then you don't even need this override at all...

**Review by abeforgit — CHANGES_REQUESTED** [review]

**Review by abeforgit — CHANGES_REQUESTED** [review]
another thing: 
image
This is supposed to say "Reglementen Van <name of administrative unit>"
if you look at the template it's pretty obvious why it's not working, it's accessing a property on the controller that doesn't exist...

File: app/routes/bestuurseenheid/reglementen/index.js
Comment (abeforgit) [reply app/routes/bestuurseenheid/reglementen/index.js:38]: actually hold off on this, the docs may be wrong

**Review by abeforgit — CHANGES_REQUESTED** [review]
And one more thing:
when we load the next 10 items in the list, there is no loading spinner or any kind of indication to the user that something is happening. So for about 4 seconds, it looks like the page freezes.

Please add a loading substate as per https://guides.emberjs.com/release/routing/loading-and-error-substates/

This will take care of the issue, but it can be quite jarring when you're technically staying on the same page, like when you request the next page of a paginated list, like here.

for that case, we can add the following to the route:
```js
  @action
  loading(transition) {
    // see snippet in https://api.emberjs.com/ember/3.27/classes/Route/events/loading?anchor=loading
    // eslint-disable-next-line ember/no-controller-access-in-routes
    const controller = this.controllerFor(this.routeName);
    if (controller && transition.from?.name === transition.to?.name) {
      controller.set('isLoadingModel', true);
      transition.promise.finally(() => {
        controller.set('isLoadingModel', false);
      });
      return false;
    }
    return true;
  }
  ```
  
  This will enhance the controller with an isLoadingModel property (which you should make `@tracked`) for those cases where the controller already exists when the model is loading. You can then use that property to toggle some css

File: app/controllers/bestuurseenheid/reglementen/index.js
Comment (usrtim) [reply app/controllers/bestuurseenheid/reglementen/index.js:7]: yes I know that, I just wanted to be more descriptive and write that on my model and then get extracts from model. Sorry for naming it wrong

File: app/routes/bestuurseenheid/reglementen/index.js
Comment (usrtim) [reply app/routes/bestuurseenheid/reglementen/index.js:38]: hmm based on you comment below you told me to move the logic of fetchi the extracts from controller to model hook, on the route(or maybe I misunderstood that ?). This is the main reason

File: app/routes/bestuurseenheid/reglementen/index.js
Comment (usrtim) [reply app/routes/bestuurseenheid/reglementen/index.js:54]: u are right, I missed something during refactoring part. Good catch

**Comment by usrtim** [comment]
> And one more thing: when we load the next 10 items in the list, there is no loading spinner or any kind of indication to the user that something is happening. So for about 4 seconds, it looks like the page freezes.
> 
> Please add a loading substate as per https://guides.emberjs.com/release/routing/loading-and-error-substates/
> 
> This will take care of the issue, but it can be quite jarring when you're technically staying on the same page, like when you request the next page of a paginated list, like here.
> 
> for that case, we can add the following to the route:
> 
> ```js
>   @action
>   loading(transition) {
>     // see snippet in https://api.emberjs.com/ember/3.27/classes/Route/events/loading?anchor=loading
>     // eslint-disable-next-line ember/no-controller-access-in-routes
>     const controller = this.controllerFor(this.routeName);
>     if (controller && transition.from?.name === transition.to?.name) {
>       controller.set('isLoadingModel', true);
>       transition.promise.finally(() => {
>         controller.set('isLoadingModel', false);
>       });
>       return false;
>     }
>     return true;
>   }
> ```
> 
> This will enhance the controller with an isLoadingModel property (which you should make `@tracked`) for those cases where the controller already exists when the model is loading. You can then use that property to toggle some css

Done

**Comment by abeforgit** [comment]
@nvdk could you check if the rdfa is semi-sensible?

**Review by elpoelma — APPROVED** [review]
Seems to work well overall, starting from an empty database.

**Pull request merged**
